packages/desktop-client/src/hooks/useCategoryMutations.ts

```diff
@@ -0,0 +1,234 @@
+import { useTranslation } from 'react-i18next';
+
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { v4 as uuidv4 } from 'uuid';
+
+import { send } from 'loot-core/platform/client/fetch';
+import {
+  type CategoryEntity,
+  type CategoryGroupEntity,
+} from 'loot-core/types/models';
+
+import { categoryQueries } from '@desktop-client/budget';
+import {
+  addGenericErrorNotification,
+  addNotification,
+} from '@desktop-client/notifications/notificationsSlice';
+import { useDispatch } from '@desktop-client/redux';
+
+type CreateCategoryPayload = {
+  name: CategoryEntity['name'];
+  groupId: CategoryGroupEntity['id'];
+  isIncome: boolean;
+  isHidden: boolean;
+};
+
+type UpdateCategoryPayload = {
+  category: CategoryEntity;
+};
+
+type DeleteCategoryPayload = {
+  id: CategoryEntity['id'];
+  transferId?: CategoryEntity['id'] | null;
+};
+
+type MoveCategoryPayload = {
+  id: CategoryEntity['id'];
+  groupId: CategoryGroupEntity['id'];
+  targetId: CategoryEntity['id'] | null;
+};
+
+type CreateCategoryGroupPayload = {
+  name: CategoryGroupEntity['name'];
+};
+
+type UpdateCategoryGroupPayload = {
+  group: CategoryGroupEntity;
+};
+
+type MoveCategoryGroupPayload = {
+  id: CategoryGroupEntity['id'];
+  targetId: CategoryGroupEntity['id'] | null;
+};
+
+type DeleteCategoryGroupPayload = {
+  id: CategoryGroupEntity['id'];
+  transferId?: CategoryGroupEntity['id'] | null;
+};
+
+export function useCategoryMutations() {
```

Comment:
Is there a particular reason to declare these all in one hook rather than provide individual hooks for each mutation? Seems like a lot of extra code run in components that don't use all mutations